cassandra-analytics-integration-tests/src/test/java/org/apache/cassandra/testing/SimpleCassandraVersion.java

```diff
@@ -0,0 +1,172 @@
+ * We want to list an adapter as being compatible with 4.0 - and that should include 4.0 alpha, etc.
+ */
+public class SimpleCassandraVersion implements Comparable<SimpleCassandraVersion>
+{
+    /**
+     * note: 3rd group matches to words but only allows number and checked after regexp test.
+     * this is because 3rd and the last can be identical.
+     **/
+    private static final String VERSION_REGEXP = "(\\d+)\\.(\\d+)(?:\\.(\\w+))?(\\-[.\\w]+)?([.+][.\\w]+)?";
+
+    private static final Pattern PATTERN = Pattern.compile(VERSION_REGEXP);
+    private static final String SNAPSHOT = "-SNAPSHOT";
+
+    public final int major;
+    public final int minor;
+    public final int patch;
+
+    /**
+     * Parse a version from a string.
+     *
+     * @param version the string to parse
+     * @return the {@link SimpleCassandraVersion} parsed from the {@code version} string
+     * @throws IllegalArgumentException if the provided string does not
+     *                                  represent a version
+     */
+    public static SimpleCassandraVersion create(String version)
+    {
+        String stripped = version.toUpperCase().replace(SNAPSHOT, "");
+        Matcher matcher = PATTERN.matcher(stripped);
+        if (!matcher.matches())
+        {
+            throw new IllegalArgumentException("Invalid Cassandra version value: " + version);
+        }
+
+        try
+        {
+            int major = Integer.parseInt(matcher.group(1));
+            int minor = Integer.parseInt(matcher.group(2));
+            int patch = matcher.group(3) != null ? Integer.parseInt(matcher.group(3)) : 0;
+
+            return SimpleCassandraVersion.create(major, minor, patch);
+        }
+        catch (NumberFormatException e)
+        {
+            throw new IllegalArgumentException("Invalid Cassandra version value: " + version, e);
+        }
+    }
+
+    public static SimpleCassandraVersion create(int major, int minor, int patch)
+    {
+        if (major < 0 || minor < 0 || patch < 0)
+        {
+            throw new IllegalArgumentException();
+        }
+        return new SimpleCassandraVersion(major, minor, patch);
+    }
+
+
+    public SimpleCassandraVersion(int major, int minor, int patch)
+    {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+    }
+
+
+    public int compareTo(SimpleCassandraVersion other)
+    {
+        if (major < other.major)
+        {
+            return -1;
+        }
+        if (major > other.major)
+        {
+            return 1;
+        }
+
+        if (minor < other.minor)
+        {
+            return -1;
+        }
+        if (minor > other.minor)
+        {
+            return 1;
+        }
+
+        if (patch < other.patch)
+        {
+            return -1;
+        }
+        if (patch > other.patch)
+        {
+            return 1;
+        }
+        return 0;
```

Comment:
nit: a more succinct version

```suggestion
        if (major != other.major)
        {
            return major - other.major;
        }
        else if (minor != other.minor)
        {
            return minor - other.minor;
        }
        else
        {
            return patch - other.patch;
        }
```